internal/ssh/ssh.go

```diff
@@ -48,23 +57,34 @@ type Config struct {
 }
 
 // NewJumpServer creates the jump server struct.
-func NewJumpServer(ctx context.Context, config Config, resolver Resolver) (Server, error) {
+func NewJumpServer(ctx context.Context, config Config, sshManager SSHManager) (Server, error) {
 	zapctx.Info(ctx, "NewJumpServer")
 
-	if resolver == nil {
+	if sshManager == nil {
 		return Server{}, fmt.Errorf("Cannot create JumpSSHServer with a nil resolver.")
 	}
 	server := Server{
 		Server: &ssh.Server{
 			Addr: fmt.Sprintf(":%s", config.Port),
 			ChannelHandlers: map[string]ssh.ChannelHandler{
-				"direct-tcpip": directTCPIPHandler(resolver),
+				"direct-tcpip": directTCPIPHandler(sshManager),
 			},
 			PublicKeyHandler: func(ctx ssh.Context, key ssh.PublicKey) bool {
+				claimUser := ctx.User()
+				if ok, err := sshManager.VerifyPublicKey(ctx, claimUser, key.Marshal()); !ok || err != nil {
```

Comment:
This logic could live purely on the ssh manager struct and be nicely encapsulated, easier to test too then.

also shouldn't it really be called sshauthoriser? it's not managing anything